### PR TITLE
build: update bazel setup to latest dev-infra and rules_nodejs version

### PR DESCRIPTION
**Pull request by devversion**
Updates the Bazel setup to the latest dev-infra and rules_nodejs
version. This includes:

* The removal of the spec-bundling code as that has been moved over to
  the shared dev-infra code.

* The remoal of the AMD esbuild and devmode JS source extraction as
  these rules have been moved to the shared dev-infra logic.

* The removal of the Angular linker ESBuild plugin creation as this
  logic is now exposed as shared script in the shared dev-infra package.

### CODE REVIEW

File: tools/defaults.bzl [hunk 1]
```diff
@@ -400,3 +405,57 @@ def ng_web_test_suite(deps = [], static_css = [], exclude_init_script = False, *
         bootstrap = bootstrap,
         **kwargs
     )
+
+def spec_bundle(name, deps, **kwargs):
```
Comment (josephperrott) [inline tools/defaults.bzl:409]: Is there a compelling reason not to have this extraction occur in `spec_bundle` from `dev-infra`? Or even as a boolean attr?  I can't think of a situation where we would want to `spec_bundle` and build both.

**Review by josephperrott — APPROVED** [review]
LGTM

Just the one comment which is more of a entire structure question that would likely be followup or at least a different scope.

File: tools/defaults.bzl (hunk 1 above)
Comment (devversion) [reply tools/defaults.bzl:409]: The issue is that the COMP repo is the only repo building devmode as ESM currently, so this does not apply at all to other repositories.

A flag would work, but I'd rather prefer to keep it local until we have decided on a plan for combining devmode & prodmode. does that sound reasonable?

Comment (josephperrott) [reply tools/defaults.bzl:409]: Ahh yes, that makes sense.  Yeah, we can just leave it as is and once and address it once we have the larger plan in place.

**Comment by zarend** [comment]
Targeted `minor` due to merge conflict with `13.0.x.`. Please create another PR if this needs to go into patch as well.

**Pull request merged**